### ISSUE TITLE
Expose queryKeyHashFn

### ISSUE THREAD

**Issue by zek** [liaoliao666/react-query-kit#10]
Can we expose queryKeyHashFn so we can just access it like ```usePost.queryKeyHashFn```

```js
const usePost = createQuery<Response, Variables, Error>({
  primaryKey: '/posts',
  queryKeyHashFn: () => 'some-custom-hash-fn'
  queryFn: ({ queryKey: [primaryKey, variables] }) => {
    // primaryKey equals to '/posts'
    return fetch(`${primaryKey}/${variables.id}`).then(res => res.json())
  },
})
```

**Comment by liaoliao666** [comment]
https://github.com/liaoliao666/react-query-kit/releases/tag/1.3.2
DONE

**Comment by liaoliao666** [comment]
Sorry for reopen. I'm currently reconsidering whether exposing queryKeyHashFn is the right thing to do. Because people will not get correct queryHashFn when they setting global queryKeyHashFn via `QueryClient`. As follows.
```js
new QueryClient({
  defaultOptions: {
    queries: {
      queryKeyHashFn: (queryKey) => ...
    }
  }
})

const usePost  = createQuery(...)

// This will return a wrong queryKeyHash because this exposed queryKeyHashFn is not aware of QueryClient
usePost.queryKeyHashFn()
```

Maybe it would be better to put this fn in returns .
```js
const usePost  = createQuery(...)

// this fn can return a correct queryKeyHash
const { queryKeyHashFn } = usePost()
```

**Comment by zek** [comment]
Instead of removing it, maybe we can just return undefined when it's not defined.

**Comment by liaoliao666** [comment]
Respect your sugestion. If queryKeyHashFn is not passed into usePost, I throw an error when people call usePost.queryKeyHashFn(). Maybe this is a good workaround😂.

**Comment by zek** [comment]
In my case, I define each query seperately and use a custom queryKeyHashFn for requests from my table component. 
With my custom queryKeyHashFn, I don't need to worry about variable type (int or string), the order etc.
My example queryKey is /api/services?page=1&sort=abc&filter[a]=1&filter=[b]=3

<img width="657" alt="image" src="https://user-images.githubusercontent.com/3463291/215269538-98e20028-5a3f-42c2-afe6-4b882422fdf9.png">

I use same query on both Client Side & Server Side (SSR). 

<img width="741" alt="image" src="https://user-images.githubusercontent.com/3463291/215269499-e292a3cf-7f94-4a57-838b-1832ca4ee512.png">

`withSsr` generates `dehydratedState` for given queries. It needs the correct queryKeyHash to generate `dehydratedState` 

<img width="463" alt="image" src="https://user-images.githubusercontent.com/3463291/215269753-40c507aa-df20-4c4a-be33-90e68f775f20.png">

I must give `queryKeyHashFn` to `prefetchQuery` to preload this query.

**Comment by zek** [comment]
I don't think throwing an error is a good idea. It will be hard to check if it's defined or not.

We can just return defined queryKeyHashFn if exists, otherwise undefined.

**Comment by liaoliao666** [comment]
you are right. I will follow your idea to solve this problem.

**Comment by liaoliao666** [comment]
Available in 1.3.6
https://github.com/liaoliao666/react-query-kit/releases/tag/1.3.6
